frontend/src/modals/ParticipantDataModal/ParticipantDataModal.tsx

```diff
@@ -0,0 +1,672 @@
+import { Filter, FilterConfigNumber, Participant } from "../../types";
+import { v4 as uuid } from "uuid";
+
+import filtersData from "../../filters_data.json";
+
+// Loading filters data before the component renders, because the Select component needs value.
+const testData = filtersData.SESSION;
+
+// We set the 'selectedFilter' to a default filter type, because the MUI Select component requires a default value when the page loads.
+const defaultFilterName = "FILTER_API_TEST";
+
+const getIndividualFilters = () => {
+  return testData.filter((filter) => filter.groupFilter !== true);
+};
+
+const getGroupFilters = () => {
+  return testData.filter((filter) => filter.groupFilter === true);
+};
+
+type Props = {
+  originalParticipant: Participant;
+  sessionId: string;
+  index: number;
+  showParticipantInput: boolean;
+  setShowParticipantInput: React.Dispatch<React.SetStateAction<boolean>>;
+  onDeleteParticipant: (index: number) => void;
+  handleParticipantChange: (index: number, participant: Participant) => void;
+  handleCanvasPlacement: () => void;
+  setSnackbarResponse: React.Dispatch<
+    React.SetStateAction<{
+      newParticipantInputEmpty: boolean;
+      requiredInformationMissing: boolean;
+      participantOriginalEmpty: boolean;
+      newInputEqualsOld: boolean;
+    }>
+  >;
+};
+
+function ParticipantDataModal({
+  originalParticipant,
+  sessionId,
+  index,
+  showParticipantInput,
+  setShowParticipantInput,
+  handleParticipantChange,
+  onDeleteParticipant,
+  setSnackbarResponse,
+  handleCanvasPlacement
+}: Props) {
+  const [participantCopy, setParticipantCopy] = useState(originalParticipant);
+  const [selectedFilter, setSelectedFilter] = useState<Filter>(
+    testData.find((filter: Filter) => filter.name === defaultFilterName)
+  );
+  const individualFilters = getIndividualFilters();
+  const groupFilters = getGroupFilters();
+  const [snackbar, setSnackbar] = useState(initialSnackbar);
+  // Setting these snackbar response values to display the notification in Session Form Page.
+  // These notifications cannot be displayed in this file, since on closing the Participant Modal,
+  // this component and the immediate parent are deleted -> hence sending the snackbar responses
+  // up to the grandparent.
+  const snackbarResponse = {
+    newParticipantInputEmpty: false,
+    requiredInformationMissing: false,
+    participantOriginalEmpty: false,
+    newInputEqualsOld: false
+  };
+
+  const handleChange = <T extends keyof Participant>(
+    objKey: T,
+    objValue: Participant[T]
+  ) => {
+    const newParticipantData = { ...participantCopy };
+    newParticipantData[objKey] = objValue;
+    setParticipantCopy(newParticipantData);
+  };
+
+  const handleFilterChange = <T extends keyof Participant>(
+    index: number,
+    key: string,
+    value: number | string,
+    keyParticipantData: T
+  ) => {
+    const filtersCopy: any = structuredClone(
+      participantCopy[keyParticipantData]
+    );
```

Comment:
I think we can upgrade the Typescript version of the project. It's 4.6.4 and structuredClone is not supported, you have to install Typescript globally and tell IDE to use that one. Updating the version (to > 4.7.x) will prevent this issue.

Reply:
The version should be already 4.9.5, we can also change it to 4.7.4. This seems to be the best minimal version, after 4.6.4